### PR TITLE
07 routing app

### PR DESCRIPTION
**Pull request by Pratik33**

### CODE REVIEW

File: 07-routing-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  const [filteredList, setFilteredList] = useState([]);
+
+  const [isMemberPresent, setIsMemberPresent] = useState(true);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchData() {
+    const data = await Promise.all(
```
Comment (Pujarini) [inline 07-routing-app/src/App.js]: push this logic into a function

File: 07-routing-app/src/component/CardComponent.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          <h4 className='card-text'>{name || login}</h4>
+          <h6>
+            Followers :{followers} | Following : {following}
+          </h6>
+          <p>
+            <small>{company || 'Student'}</small>
+          </p>
+        </div>
+        <div className='cardFooter text-center row m-0 p-0'>
+          <div className='col-4 m-auto'>
```
Comment (Pujarini) [inline 07-routing-app/src/component/CardComponent.js:33]: create a reusable component for the icons

**Review by Pujarini — CHANGES_REQUESTED** [review]

File: 07-routing-app/src/App.js [hunk 3]
```diff
@@ -40,18 +41,37 @@ const AppLayoutComponent = () => {
...
+      <Outlet />
     </>
   );
 };
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 
-root.render(<AppLayoutComponent />);
+const router = createBrowserRouter([
+  {
```
Comment (Pujarini) [inline 07-routing-app/src/App.js:33]: push this into constant

**Review by Pujarini — CHANGES_REQUESTED** [review]

**Comment by Pujarini** [comment]
reviewed. 🚀

File: 07-routing-app/src/component/CardComponent.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            <a
+              target='blank'
+              href={html_url}>
+              <i className='fa-brands fa-twitter'></i>
+            </a>
+          </div>
+          <div className='col-4 m-auto'>
+            <a
+              target='blank'
```
Comment (ap221882) [inline 07-routing-app/src/component/CardComponent.js:49]: It is `target = "_blank"`

File: 07-routing-app/src/component/CardHolder.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Link } from 'react-router-dom';
+import CardComponent from './CardComponent';
+
+const CardHolder = ({ filteredTeam }) => {
+  if (filteredTeam.length > 0) {
+    return filteredTeam.map((user) => {
+      if (user.login) {
+        return (
+          <Link to={`${user.login}`}>
```
Comment (ap221882) [inline 07-routing-app/src/component/CardHolder.js:9]: give key attribute to Link not Card

File: 07-routing-app/src/component/NavbarComponent.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Link } from 'react-router-dom';
+import SearchComponent from './SearchComponent';
+const NavbarComponent = ({ data, setFilteredList, setIsMemberPresent }) => {
+  return (
+    <nav className='navbar navbar-light '>
```
Comment (ap221882) [inline 07-routing-app/src/component/NavbarComponent.js:5]: Use prettier. There is one extra space after `light`

File: 07-routing-app/src/component/SearchComponent.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+
+const searchLogin = (searchKey, data) => {
+  return data.filter((team) => {
+    return (
+      team?.name?.toLowerCase().includes(searchKey.toLowerCase()) ||
+      team?.company?.toLowerCase().includes(searchKey.toLowerCase()) ||
```
Comment (ap221882) [inline 07-routing-app/src/component/SearchComponent.js:7]: make this a reusable function.

File: 07-routing-app/src/component/SearchComponent.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      team?.login?.toLowerCase().includes(searchKey.toLowerCase())
+    );
+  });
+};
+const SearchComponent = ({ data, setFilteredList, setIsMemberPresent }) => {
+  const [searchKey, setSearchKey] = useState('');
+  return (
+    <form
+      className='searchForm'
+      onSubmit={(e) => {
```
Comment (ap221882) [inline 07-routing-app/src/component/SearchComponent.js:17]: keep logic in separate handler function

File: 07-routing-app/src/component/TeamMemberDetails.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+
+import CardComponent from './CardComponent';
+
+const TeamMemberDetails = () => {
+  const { teamMemberId } = useParams();
+  const [user, setUser] = useState({});
+  useEffect(() => {
+    const data = fetchUserDetails();
```
Comment (ap221882) [inline 07-routing-app/src/component/TeamMemberDetails.js:12]: You're not using this `data` variable. Remove it.

File: 07-routing-app/src/component/TeamMemberDetails.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  }, []);
+  const fetchUserDetails = async () => {
+    const data = await fetch(`https://api.github.com/users/${teamMemberId}`);
```
Comment (ap221882) [inline 07-routing-app/src/component/TeamMemberDetails.js:15]: handle error-handling logic inside async function using try-catch.

File: 07-routing-app/src/component/TeamMembersComponent.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
+import CardHolder from './CardHolder';
+import SearchComponent from './SearchComponent';
+
+import { useState, useEffect } from 'react';
```
Comment (ap221882) [inline 07-routing-app/src/component/TeamMembersComponent.js:4]: Have global imports on top

File: 07-routing-app/public/index.html [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+      href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css"
+      integrity="sha384-9gVQ4dYFwwWSjIDZnLEWnxCjeSWFphJiwGPXr1jddIhOegiu1FwO5qRGvFXOdJZ4"
+      crossorigin="anonymous" />
+  </head>
+
+  <body>
+    <div id="root"></div>
+    <script
+      type="module"
+      src="../src/App.js"></script>
```
Comment (ap221882) [inline 07-routing-app/public/index.html:21]: Have comments that which script is for which usecase

**Review by ap221882 — COMMENTED** [review]
Nicely coded. Kudos!

File: 07-routing-app/src/component/NavbarComponent.js (hunk 6 above)
Comment (Pratik33) [reply 07-routing-app/src/component/NavbarComponent.js:5]: Removed

File: 07-routing-app/src/component/TeamMemberDetails.js (hunk 9 above)
Comment (Pratik33) [reply 07-routing-app/src/component/TeamMemberDetails.js:12]: removed

File: 07-routing-app/src/component/TeamMembersComponent.js (hunk 11 above)
Comment (Pratik33) [reply 07-routing-app/src/component/TeamMembersComponent.js:4]: arranged properly

**Pull request merged**